### ISSUE TITLE
[Bug] Actor scheduler with fractional GPUs

### ISSUE THREAD

**Issue by etiennedub** [ray-project/ray#23492]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Serve

### Issue Severity

High: It blocks me to complete my task.

### What happened + What you expected to happen

When trying to deploy a Deployment from Ray serve with fractional `num_gpus` sometime the scheduler cannot deploy all the replicas even if enough resources are available. It seems to depend on the number `num_gpus` and `num_replicas`.

### Versions / Dependencies

Python 3.8.12

ray[serve]==1.11.0

### Reproduction script

```
import ray
import time

from ray import serve

@serve.deployment(num_replicas=6, ray_actor_options={"num_gpus":0.6, "num_cpus":0})
def ml_model(data):
    print("Model")
    time.sleep(1)
    return 1

ray.init(address="ray://localhost:10001", namespace="serve")
serve.start(detached=True, http_options={
    "host":"0.0.0.0"
    })

print(ray.available_resources())
ml_model.deploy()
```

Output:
```
(ServeController pid=2713) 2022-03-25 12:56:23,516      INFO http_state.py:98 -- Starting HTTP proxy with name 'SERVE_CONTROLLER_ACTOR:SERVE_PROXY_ACTOR-node:10.128.15.229-0' on node 'node:10.128.15.229-0' listening on '0.0.0.0:8000'
(HTTPProxyActor pid=2756) INFO:     Started server process [2756]
2022-03-25 08:56:25,889 INFO api.py:521 -- Started detached Serve instance in namespace 'serve'.
{'object_store_memory': 30478928400.0, 'accelerator_type:T4': 3.0, 'GPU': 6.0, 'node:10.128.15.230': 1.0, 'CPU': 24.0, 'memory': 70365090203.0, 'node:10.128.15.231': 1.0, 'node:10.128.15.229': 0.98, 'node:10.128.15.232': 1.0}
2022-03-25 08:56:26,098 INFO api.py:262 -- Updating deployment 'ml_model'. component=serve deployment=ml_model
(ServeController pid=2713) 2022-03-25 12:56:26,094      INFO deployment_state.py:920 -- Adding 6 replicas to deployment 'ml_model'. component=serve deployment=ml_model
The actor or task with ID ffffffffffffffffe477d1125d31896b030338ed12000000 cannot be scheduled right now. You can ignore this message if this Ray cluster is expected to auto-scale or if you specified a runtime_env for this actor or task, which may tak
e time to install.  Otherwise, this is likely due to all cluster resources being claimed by actors. To resolve the issue, consider creating fewer actors or increasing the resources available to this Ray cluster.
Required resources for this actor or task: {GPU: 0.600000}
Available resources on this node: {8.000000/8.000000 CPU, 20.442487 GiB/20.442487 GiB memory, 0.800000/2.000000 GPU, 8.761066 GiB/8.761066 GiB object_store_memory, 1.000000/1.000000 accelerator_type:T4, 1.000000/1.000000 node:10.128.15.231}
 In total there are 0 pending tasks and 2 pending actors on this node.
(ServeController pid=2713) 2022-03-25 12:56:56,222      WARNING deployment_state.py:1117 -- Deployment 'ml_model' has 2 replicas that have taken more than 30s to be scheduled. This may be caused by waiting for the cluster to auto-scale, or waiting for
 a runtime environment to install. Resources required for each replica: {'GPU': 0.6}, resources available: {'GPU': 3.6}. component=serve deployment=ml_model
```

### Anything else

It depends on the `num_replicas` value and `num_gpus` value.

For example, with 6 replicas and 0.5 or 1 GPU per replica it deploys without issue but with 0.55 or 0.6 GPU per replica the scheduler complaint.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by simon-mo** [comment]
This is because Ray cannot provision an actor that uses fractional resource to span across two GPUs. 

For example, say each replica requires `{"GPU": 0.6}`. And total of 2 available GPUs. Theoretically it might look like we can fit 3 replicas because `0.6*3 < 2`. But Ray can't allocate the third one like that. Once the first two replicas assigned, it will look like first GPU has 0.6 allocated, and the second has 0.6 allocated. There is no integral GPU to allocate to the third replica. Afterall, you can't assign a 0.6 GPU to the 0.4 and 0.4 remaining GPU compute.

Please let me know if this explanation is messy.

**Comment by etiennedub** [comment]
Maybe I don't understand your explanation, but in your example, the number of replicas is higher than the number of GPUs available. In my case, I have 6 GPUs and 6 replicas. Also, if I set `{"GPU": 1}` all my replicas are allocated, but with `{"GPU": 0.6}` they are not. From what I understand, it should work with every possibility if GPU < 1.

Let me know If I don't understand correctly your explanation.

**Comment by simon-mo** [comment]
I see. Sorry for the misunderstanding. If that's the case this is definitely a bug.

One clue is this `'accelerator_type:T4': 3.0, 'GPU': 6.0` in available resource line. Can you show the `nvidia-smi` output by any chance? Is there any other workload also consuming this acclerator?

**Comment by etiennedub** [comment]
Sorry, it is hard for me to reproduce at this time. When I did my tests, the 6 GPUs were delegate to Ray and no other process were running. Also, GPU memory was not an issue.

I also note that the number of `accelerator_type:T4` doesn't match the number of GPUs, when there are multiple GPUs per VM. I'm not really sure what it implied or if it's a bug.
